src/main/java/seedu/address/model/DeliveryBook.java

```diff
@@ -86,13 +87,25 @@ public void removeDelivery(Delivery key) {
         deliveries.remove(key);
     }
 
+    /**
+     * Removes all deliveries associated with the given customer
+     *
+     * @param customer The customer whose deliveries are to be removed
+     */
+    public void removeDeliveryByCustomer(Customer customer) {
+        // get id of customer, delete deliveries with matching customer id
+        int key = customer.getCustomerId();
+        // concurrent modification exception
+        deliveries.removeIf(delivery -> delivery.getCustomer().getCustomerId() == key);
```

Comment:
Would it be possible to get the customer id from delivery directly, so as to abide by LoD

Reply:
Good spot, done!